include/pybind11/cast.h

```diff
@@ -495,13 +495,20 @@ template <typename T> typename make_caster<T>::template cast_op_type<T> cast_op(
     return cast_op<T>(caster);
 }
 
-template <typename type> class type_caster<std::reference_wrapper<type>> : public type_caster_base<type> {
+template <typename type> class type_caster<std::reference_wrapper<type>> {
+private:
+    using caster_t = make_caster<type>;
+    caster_t subcaster;
 public:
+    bool load(handle src, bool convert) { return subcaster.load(src, convert); }
+    static PYBIND11_DESCR name() { return caster_t::name(); }
     static handle cast(const std::reference_wrapper<type> &src, return_value_policy policy, handle parent) {
-        return type_caster_base<type>::cast(&src.get(), policy, parent);
+        return caster_t::cast(&src.get(), policy, parent);
```

Comment:
So I agree that `rvp::automatic` and `take_ownership` seem wrong and should be masked, but `copy` could be valid.  Perhaps it would be better to just set it to `rvp::reference` if it's set to `automatic` or `take_ownership`, and otherwise leave it alone?